lib/seattleflu/id3c/cli/command/etl/clinical.py

```diff
@@ -104,45 +115,266 @@ def etl_clinical(*, db: DatabaseSession):
+    site_identifier = create_identifier(f"{SFS}/site", encounter_site)
+    site_reference = create_reference(
+        reference_type = "Location",
+        identifier = site_identifier
+    )
+
+    location_references = resident_locations or []
+    location_references.append(site_reference)
+
+    return list(map(lambda ref: {"location": ref}, location_references))
 
-            encounter = upsert_encounter(db,
-                identifier      = record.document["identifier"],
-                encountered     = record.document["encountered"],
-                individual_id   = individual.id,
-                site_id         = site.id,
-                age             = age(record.document),
-                details         = encounter_details(record.document))
 
-            sample = update_sample(db,
-                sample = sample,
-                encounter_id = encounter.id)
+def create_encounter(db: DatabaseSession,
+                     record: dict,
+                     patient_reference: dict,
+                     location_references: list) -> Optional[tuple]:
+    """ Returns a FHIR Encounter resource entry and reference """
+    encounter_location_references = create_encounter_location_references(db, record, location_references)
 
-            # Link encounter to a Census tract, if we have it
-            tract_identifier = record.document.get("census_tract")
+    if not encounter_location_references:
+        return None, None
 
-            if tract_identifier:
-                # Special-case float-like identifiers in earlier date
-                tract_identifier = re.sub(r'\.0$', '', str(tract_identifier))
+    encounter_date = record["encountered"]
+    if not encounter_date:
+        return None, None
 
-                tract = find_location(db, "tract", tract_identifier)
-                assert tract, f"Tract «{tract_identifier}» is unknown"
+    encounter_id = record["identifier"]
+    encounter_identifier = create_identifier(f"{SFS}/encounter", encounter_id)
+
+    encounter_class = create_encounter_class(record)
+    encounter_status = create_encounter_status(record)
+
+    record_source = f"{record['_provenance']['filename']},row:{record['_provenance']['row']}"
+
+    encounter_resource = create_encounter_resource(
+        encounter_source = record_source,
+        encounter_identifier = [encounter_identifier],
+        encounter_class = encounter_class,
+        encounter_date = encounter_date,
+        encounter_status = encounter_status,
+        patient_reference = patient_reference,
+        location_references = encounter_location_references,
+    )
+
+    return create_entry_and_reference(encounter_resource, "Encounter")
+
+
+def create_resident_locations(record: dict) -> Optional[tuple]:
+    """
+    Returns FHIR Location resource entry and reference for resident address
+    and Location resource entry for Census tract.
+    """
+    if not record["address"]:
```

Comment:
Do we expect the address itself to be included at this level, or just the census tract?

Reply:
The key name is a little misleading here, but the `address` in the ndjson file is already hashed so that's what's being ingested here. That could be renamed to `address_hash` during parsing.